test_engine/terraform/aws/ubuntu/main.tf

```diff
@@ -0,0 +1,321 @@
+    Name = "build_engine_aws_private_rt-${random_string.random_suffix.id}"
+  }
+}
+
+# Assciate public subnet to public route table
+resource "aws_route_table_association" "build_engine_aws_public_subnet_rt_association" {
+  depends_on = [
+    aws_subnet.build_engine_aws_public_subnet,
+    aws_route_table.build_engine_aws_public_rt
+  ]
+
+  subnet_id      = aws_subnet.build_engine_aws_public_subnet.id
+  route_table_id = aws_route_table.build_engine_aws_public_rt.id
+}
+
+# Assciate private subnet to private route table
+resource "aws_route_table_association" "build_engine_aws_private_subnet_rt_association" {
+  depends_on = [
+    aws_subnet.build_engine_aws_private_subnet,
+    aws_route_table.build_engine_aws_private_rt
+  ]
+
+  subnet_id      = aws_subnet.build_engine_aws_private_subnet.id
+  route_table_id = aws_route_table.build_engine_aws_private_rt.id
+}
+
+# Create AWS key pair
+resource "aws_key_pair" "build_engine_aws_pair_key" {
+  key_name   = format("%s_%s", "build_engine_aws_key_pair", "${random_string.random_suffix.id}")
+  public_key = file(var.aws_ssh_public_key_file_path)
+}
+
+/*
+# Create cluster secret (used for k3s on arm64 only)
+resource "random_password" "k3s_cluster_secret" {
+  length = var.arch == "arm64" ? 64 : 0
+  special = false
+}
+*/
+
+# Create controlplane instances
+resource "aws_instance" "build_engine_aws_instance" {
+ depends_on = [
+    aws_subnet.build_engine_aws_public_subnet,
+  ]
+
+  count = var.build_engine_aws_instance_count
+
+  availability_zone = var.aws_availability_zone
+
+  ami           = data.aws_ami.aws_ami_ubuntu.id
+  instance_type = var.build_engine_aws_instance_type
+
+  subnet_id = aws_subnet.build_engine_aws_public_subnet.id
+  vpc_security_group_ids = [
+    aws_security_group.build_engine_aws_secgrp_controlplane.id
+  ]
+
+  root_block_device {
+    delete_on_termination = true
+    volume_size = var.build_engine_aws_instance_root_block_device_size
+  }
+
+  key_name = aws_key_pair.build_engine_aws_pair_key.key_name
+  user_data = file("${path.module}/user-data-scripts/provision_amd64.sh")
+  #user_data = file("${var.tf_workspace}/terraform/aws/ubuntu/user-data-scripts/provision_amd64.sh")
+
+  tags = {
+    Name = "${var.build_engine_aws_instance_name}-${count.index}-${random_string.random_suffix.id}"
+    DoNotDelete	= "true"
+    Owner = "longhorn-infra"
+  }
+}
+
+resource "aws_eip" "build_engine_aws_eip_controlplane" {
+  count    = var.build_engine_aws_instance_count
+  vpc      = true
+}
+
+# Associate every EIP with controlplane instance
+resource "aws_eip_association" "build_engine_aws_eip_assoc" {
+  depends_on = [
+    aws_instance.build_engine_aws_instance,
+    aws_eip.build_engine_aws_eip_controlplane
+  ]
+
+  count    = var.build_engine_aws_instance_count
+
+  instance_id   = element(aws_instance.build_engine_aws_instance, count.index).id
+  allocation_id = element(aws_eip.build_engine_aws_eip_controlplane, count.index).id
+}
+
+
+output "ssh_key" {
```

Comment:
can you elaborate why do we need ssh key as output ?

Reply:
This part is not necessary, I had removed this part of code